common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -201,6 +202,28 @@ def get_static_transcript(self, request, transcripts):
                 )
         return response
 
+    @XBlock.json_handler
+    def publish_completion(self, data, dispatch):  # pylint: disable=unused-argument
+        """
+        Entry point for completion for student_view.
+
+        Parameters:
+            data: JSON dict:
+                key: "completion"
+                value: float in range [0.0, 1.0]
+
+            dispatch: Ignored.
+        Return value: JSON response (200 on success, 400 for malformed data)
+        """
+        completion_service = self.runtime.service(self, 'completion')
+        if not completion_service or not completion_service.completion_tracking_enabled():
+            raise JsonHandlerError(404, u"Completion service not available")
```

Comment:
My only problem with this is that the console logs get filled with 500 errors if the completion tracking waffle switch isn't set.  Shouldn't that case fail quietly?

Reply:
New version passes completionEnabled to the client, and the client only attaches the event handlers if completion is enabled.

Reply:
Thanks for splitting these.  I still think changing the second from "Completion service not available" to something like "Completion tracking is not enabled and api calls are unexpected" would help someone understand more quickly where to look for the problem.